examples/server/server.cpp

```diff
@@ -2490,6 +2490,7 @@ int main(int argc, char ** argv) {
 #endif
     // own arguments required by this example
     gpt_params params;
+    params.log_json = server_log_json;
```

Comment:
This code will never work.

Here you assigned `params.log_json = server_log_json;` and below we already had `server_log_json = params.log_json;`, so `server_log_json` will end up the same value no matter what.

Reply:
Sorry I overlooked this. This will still work but the code is very strange and not readable.

I'd prefer to have 2 separated variables for 2 examples (i.e. `server_log_json`vs `batched_bench_output_json`), as they don't use the same default value.

Reply:
So the reason for this is that `params.log_json` defaults to `false` when a `gpt_params` is initialized, but `server_log_json` is initialized to `true`.

We copy that global's value here so the `--help` output shows the correct default for each tool (false for `batched-bench`, true for `server`); before this, it just assumed it's JSON by default (which it really is for `server`).

I don't think `batched-bench` needs a `batched_bench_output_json`, because it just directly reads the parameter as set by the user.

Reply:
IMO setting default params outside of `gpt_params` constructor is not intuitive. All default params are supposed to be defined in one single place.

The way you do still works, but other maintainers/contributors will make the same mistake as mine.

Also, the param is still marked as used by `server`, so it doesn't give a very intuitive help message.

Reply:
> IMO setting default params outside of `gpt_params` constructor is not intuitive. All default params are supposed to be defined in one single place.

How would you suggest this be resolved? The default for `server` is `true`, but it probably shouldn't be changed for other apps – and other llama.cpp apps using `gpt_params_parse` show all of the `server` options (as well as all of the others, confusingly) even if they don't use them. 

> Also, the param is still marked as used by `server`, so it doesn't give a very intuitive help message.

Yes, I touched upon this in the PR message: "I know `--log-format` is in the `server` section". I don't think it's a great idea to move it into e.g. the `logging` section because that section is conditionally compiled (`#ifndef LOG_DISABLE_LOGS`). Ideas?

Reply:
I already gave the answer in [my comment above](https://github.com/ggerganov/llama.cpp/pull/9293#discussion_r1741998599).

> I know --log-format is in the server section, but this seems pretty harmless...

No, not harmless. It harms the readability of the code.